app/controllers/application_controller/ci_processing.rb

```diff
@@ -1760,6 +1760,27 @@ def process_hosts(hosts, task, display_name = nil)
           add_flash(_("\"%{task}\": not supported for %{hostname}") % {:hostname => host.name, :task => (task_name || task)}, :error)
         end
       end
+    when "maintenance"
+      each_host(hosts, task_name) do |host|
+        if host.maintenance
+          if host.respond_to?(:unset_node_maintenance)
+            host.send(:unset_node_maintenance)
+            add_flash(_("\"%{record}\": %{task} successfully initiated") %
+                      {:record => host.name, :task => (display_name || task)})
+          else
+            add_flash(_("\"%{task}\": not supported for %{hostname}") %
+                      {:hostname => host.name, :task => (task_name || task)}, :error)
+          end
+        elsif host.respond_to?(:set_node_maintenance)
+          host.send(:set_node_maintenance)
+          add_flash(_("\"%{record}\": %{task} successfully initiated") %
+                    {:record => host.name, :task => (display_name || task)})
+        else
+          add_flash(_("\"%{task}\": not supported for %{hostname}") %
+                    {:hostname => host.name, :task => (task_name || task)}, :error)
+        end
+      end
+      Host.refresh_ems(hosts)
```

Comment:
The refresh feels wrong here, particularly since it's in generic UI code.  I would think we would get the corresponding event, and the regular event handler would kick off its normal refresh.